### DISCUSSION TITLE
Bms 3.0 seplos

### DISCUSSION THREAD

**Discussion by jolly12f** [syssi/esphome-seplos-bms#63] [General]
Hello, is this compatible with new generation bms?

**Comment by syssi** [comment]
I don't know. Could you ask the manufacturer about protocol details / specs?

**Reply by jolly12f** [reply]
If you tell me exactly what to ask I will, I'm not very practical

**Reply by syssi** [reply]
"Dear customer support, could you provide a communication protocol description applicable for your new Seplos BMS v3.0?"

**Comment by jolly12f** [comment]
![Screenshot_2023-08-10-10-27-02-306_cn wps xiaomi abroad lite](https://github.com/syssi/esphome-seplos-bms/assets/124673918/a2d3e5d5-d1d1-4a53-a761-25b79197269c)

**Reply by jolly12f** [reply]
That's enough, I have 19 pages

**Reply by syssi** [reply]
Could you provide the complete document? It looks different to the Seplos-ASCII protocol. May be it's Modbus.

**Reply by jolly12f** [reply]
If you want I can send you the whole document

**Reply by jolly12f** [reply]
https://drive.google.com/file/d/1O-ETwGw6yGQaGapOMh9V9eGrp1vwGR1j/view?usp=drivesdk

**Reply by syssi** [reply]
It's ModbusRTU. You could adapt this project easily: https://github.com/syssi/esphome-pace-bms

This project supports the Seplos-ASCII protocol only and is incompatible.

**Reply by jolly12f** [reply]
I don't understand what I have to do, in yaml configuration

**Reply by syssi** [reply]
I will provide some examples. Do you want to use a ESP32 or ESP8266?

**Reply by syssi** [reply]
Let's try something like this: https://github.com/syssi/esphome-seplos-bms/blob/add-seplos-v3-support/esp32-seplos-v3-example.yaml

**Reply by jolly12f** [reply]
I use esp8266

**Reply by syssi** [reply]
In this case please use this YAML: https://github.com/syssi/esphome-seplos-bms/blob/add-seplos-v3-support/esp8266-seplos-v3-example.yaml

And attach the RS485 converter to GPIO4/GPIO5. Does the BMS respond?

**Reply by jolly12f** [reply]
I'll try this evening, I use AZ delivery, the TX and RX pins or D1 o D2? 
![IMG_20230809_235659](https://github.com/syssi/esphome-seplos-bms/assets/124673918/b6c3abc8-5a35-497d-b5fc-b504dbb2f9e9)

**Reply by syssi** [reply]
Please use `D1` (GPIO5) and `D2` (GPIO4). RX/TX is connected to your USB-to-TTL converter and useful to monitor the log output of the ESPHome node via the micro-USB cable.

**Comment by jolly12f** [comment]
Then something I managed to upload, but the 4 entities remain unknown

**Reply by syssi** [reply]
Could you provide a photo of your wiring? Which communication port of the BMS do you use?

**Reply by jolly12f** [reply]
I solved everything there was a writing error in the yaml the register_type entry must be read now I read all the values

**Comment by jolly12f** [comment]
![Screenshot_2023-08-11-08-53-43-858_io homeassistant companion android](https://github.com/syssi/esphome-seplos-bms/assets/124673918/4cdd4f05-95ad-49be-b221-d85dbf747e78)

**Reply by syssi** [reply]
Could you contribute your YAML and some details about your wiring? I would like to extend this project. Thanks in advance!

**Reply by jolly12f** [reply]
I'm checking because it was going last night but I'm not getting any data today

**Comment by jolly12f** [comment]
now my problem is that the TTL 485 card burns out after 10 minutes and I don't understand why.

**Reply by syssi** [reply]
It's hard to provide any support of you don't share details.

**Reply by jolly12f** [reply]
Maybe I had a USB power supply that didn't fit I'm trying another one

**Comment by jolly12f** [comment]
Ok fixed everything was the power supply that was not good, I don't know how my yaml loads can you tell me? So I put it online

**Reply by syssi** [reply]
If you rename the YAML file to `.txt` the file can be attached to a comment here.

**Comment by jolly12f** [comment]
[seplos_bms_3.0.yaml.txt](https://github.com/syssi/esphome-seplos-bms/files/12325024/seplos_bms_3.0.yaml.txt)

**Reply by jolly12f** [reply]
Same wiring as for d1 mini

**Comment by jolly12f** [comment]
Do you think it is possible to read another bms using 2 other inputs??

**Reply by syssi** [reply]
Yes. This is possible.

**Reply by jolly12f** [reply]
Will I have to rewrite at the end of my yaml all the same but with different pins?

**Reply by syssi** [reply]
Some sections must be lists and you have to address to different instances using IDs. Do you own two Seplos BMS V3 devices?

**Reply by jolly12f** [reply]
I will get another battery later, I have to understand if with the master bms I can also read the slave bms otherwise I have to read them individually one by one

**Comment by jolly12f** [comment]
> Some sections must be lists and you have to address to different instances using IDs. Do you own two Seplos BMS V3 devices?



Hi everyone, I integrated my seplos bms with modbus protocol with my esp8266, when I only had one battery everything worked, now I added a second battery with a second BMS which is connected to the first via port 485, and my entire esp system has often doesn’t work, I have no idea how to reconfigure my file. can anyone give me a hand?? I attach my current configuration that works with a single battery and the description of the Modbus protocol of my BMS. Please if anyone can help me I would be grateful

this is my configuration:

esp8266:
board: esp01_1m

wifi:
ssid:…

password:…
ota:
password: “…”

logger:
level: DEBUG

api:
encryption:
key: “…”

uart:

id: uart_0
baud_rate: 19200
tx_pin: GPIO4
rx_pin: GPIO5
debug:
direction: BOTH
dummy_receiver: false
modbus:

id: modbus0
uart_id: uart_0
send_wait_time: 200ms
modbus_controller:

id: bms0
address: 0x00
modbus_id: modbus0
command_throttle: 200ms
update_interval: 10s
sensor:

platform: modbus_controller
modbus_controller_id: bms0
name: “${name} cicli carica”
address: 0x1007
register_type: read
value_type: S_WORD
unit_of_measurement: “”
device_class: battery
state_class: measurement
accuracy_decimals: 0
filters:
multiply: 1

**Comment by jolly12f** [comment]
I read that other people have the same problem as me, with just one battery the Modbus system is able to read the data, but with two batteries no one has yet managed to get a reading of the batteries?

**Comment by ferelarg** [comment]
As I answered you in the other message, you can only read using this method when you have only one battery.
Since you set the esphome as Master. In Modbus communication, there can only be one master, that is, only one device can consult registers, etc.
